humanlayer/core/protocol.py

```diff
@@ -63,8 +64,32 @@ def contacts(self) -> AdminStore[HumanContact, HumanContactStatus]:
 
 class HumanLayerException(Exception):
     @staticmethod
-    def raise_for_status(resp: requests.Response) -> None:
+    def _handle_response_error(resp: requests.Response | aiohttp.ClientResponse, e: Exception) -> None:
+        raise HumanLayerException(f"{e}: {getattr(resp, 'text', str(e))}")
+
+    @staticmethod
+    def _raise_client_error(resp: aiohttp.ClientResponse) -> "HumanLayerException":
+        e = aiohttp.ClientResponseError(
+            request_info=resp.request_info,
+            history=resp.history,
+            status=resp.status,
+        )
+        return HumanLayerException(f"{e}: {e!s}")
+
+    @staticmethod
+    def _raise_request_error(resp: requests.Response) -> "HumanLayerException":
         try:
             resp.raise_for_status()
         except requests.HTTPError as e:
-            raise HumanLayerException(f"{e}: {resp.text}") from e
+            return HumanLayerException(f"{e}: {resp.text}")
+        return HumanLayerException("Unknown error")  # Should never reach here
+
+    @staticmethod
+    def raise_for_status(resp: requests.Response | aiohttp.ClientResponse) -> "HumanLayerException":
```

Comment:
Going to ignore this for now because if we use `raise` then it throws this error
```
humanlayer/core/async_cloud.py:63: error: Exception must be derived from BaseException  [misc]
```

when running 
```
                raise HumanLayerException.raise_for_status(response)
```